File: api-net/Controllers/PublicFaultController.cs
using LiftOtonom.Api.Data;
using LiftOtonom.Api.Models;
using LiftOtonom.Api.Services;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;

namespace LiftOtonom.Api.Controllers;

/// <summary>QR ile GİRİŞSİZ arıza bildirimi (auth YOK).</summary>
[ApiController]
[Route("api/v1/public")]
public class PublicFaultController(AppDbContext db, ITenantContext tenantCtx, FaultNotificationService notify) : ControllerBase
{
    public record FaultDto(string? ReporterName, string? ReporterPhone, string Description);

    [HttpGet("qr/{qrToken}")]
    public async Task<IActionResult> Show(string qrToken)
    {
        var e = await db.Elevators.IgnoreQueryFilters()
            .Include(x => x.Building).FirstOrDefaultAsync(x => x.QrToken == qrToken && x.DeletedAt == null)
            ?? throw new ApiException(404, "Asansör bulunamadı.");
        return Ok(new
        {
            elevator = new { e.Id, e.Name, e.Brand },
            building = e.Building == null ? null : new { e.Building.Name, e.Building.Address },
        });
    }

    [HttpPost("fault-reports/{qrToken}")]
    public async Task<IActionResult> Store(string qrToken, FaultDto dto)
    {
        if (string.IsNullOrWhiteSpace(dto.Description)) throw new ApiException(422, "Açıklama zorunludur.");
        var e = await db.Elevators.IgnoreQueryFilters().FirstOrDefaultAsync(x => x.QrToken == qrToken && x.DeletedAt == null)
            ?? throw new ApiException(404, "Asansör bulunamadı.");

        tenantCtx.TenantId = e.TenantId; // kayıt doğru firmaya düşsün
        var now = DateTime.UtcNow;
        var fault = new FaultReport
        {
            TenantId = e.TenantId, ElevatorId = e.Id, ReportedByType = "qr", Priority = "normal", Status = "new",
            Description = $"{dto.ReporterName} {dto.ReporterPhone} — {dto.Description}".Trim(),
            CreatedAt = now, UpdatedAt = now,
        };
        db.FaultReports.Add(fault);
        await db.SaveChangesAsync();
        await notify.NotifyAsync(fault, FaultNotificationService.Stage.Created); // otonom WhatsApp
        return StatusCode(201, new { message = "Arıza talebiniz alındı. Ekibimiz en kısa sürede ilgilenecek.", report_id = fault.Id });
    }
}

File: api/app/Http/Controllers/Api/PublicFaultController.php
<?php
// app/Http/Controllers/Api/PublicFaultController.php

namespace App\Http\Controllers\Api;

use App\Http\Controllers\Controller;
use App\Models\Elevator;
use App\Models\FaultReport;
use App\Support\TenantContext;
use Illuminate\Http\Request;

/**
 * QR kod ile GİRİŞSİZ arıza bildirimi (v1.1 26.9).
 * Asansör qr_token'dan çözülür; tenant ona göre belirlenir.
 */
class PublicFaultController extends Controller
{
    /** QR açılış: asansör/bina bilgisi (girişsiz görünür). */
    public function show(string $qrToken)
    {
        $elevator = Elevator::withoutGlobalScopes()
            ->with('building:id,name,address')
            ->where('qr_token', $qrToken)
            ->firstOrFail();

        return response()->json([
            'elevator' => [
                'id'    => $elevator->id,
                'name'  => $elevator->name,
                'brand' => $elevator->brand,
            ],
            'building' => $elevator->building ? [
                'name'    => $elevator->building->name,
                'address' => $elevator->building->address,
            ] : null,
        ]);
    }

    /** Girişsiz arıza kaydı oluştur. */
    public function store(Request $request, string $qrToken)
    {
        $data = $request->validate([
            'reporter_name'  => ['nullable', 'string', 'max:255'],
            'reporter_phone' => ['nullable', 'string', 'max:20'],
            'description'    => ['required', 'string', 'max:2000'],
        ]);

        $elevator = Elevator::withoutGlobalScopes()
            ->where('qr_token', $qrToken)
            ->firstOrFail();

        // tenant'ı asansörden belirle ki kayıt doğru firmaya düşsün
        TenantContext::setId($elevator->tenant_id);

        $fault = FaultReport::create([
            'tenant_id'        => $elevator->tenant_id,
            'elevator_id'      => $elevator->id,
            'reported_by_type' => 'qr',
            'priority'         => 'normal',
            'status'           => 'new',
            'description'      => trim(
                ($data['reporter_name'] ?? '') . ' ' .
                ($data['reporter_phone'] ?? '') . ' — ' . $data['description']
            ),
        ]);

        return response()->json([
            'message'   => 'Arıza talebiniz alındı. Ekibimiz en kısa sürede ilgilenecek.',
            'report_id' => $fault->id,
        ], 201);
    }
}
